src/gilded_rose.js

```diff
@@ -14,49 +15,91 @@ items.push(new Item('Backstage passes to a TAFKAL80ETC concert', 15, 20));
 items.push(new Item('Conjured Mana Cake', 3, 6));
 
 function update_quality() {
+  var degradeFactor
+  var sellinDecrementor
+  var resetFactor
+
   for (var i = 0; i < items.length; i++) {
-    if (items[i].name != 'Aged Brie' && items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
-      if (items[i].quality > 0) {
-        if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
-          items[i].quality = items[i].quality - 1
-        }
+
+    degradeFactor=1
+    sellinDecrementor=1
+    resetFactor=1
+
+    if (items[i].name.includes("Conjured")) {
```

Comment:
One of the major problems with the original implementation was the use of multiple and nested `if`'s - if you have a design like this, you tend to follow this pattern when extending the system.  As we discussed in class, this approach won't lead to an "open for extension" design. 

For future exercises, when in the refactor phase, as yourselves "What can I remove from this?" Consider asking an instructor to join the pair during the refactor phase and ask them to help you identify ways to remove as much code as possible from the implementation - do this early in the implementation ( maybe 3rd or 4th passing test) so there is less to refactor.